mtags-shared/src/main/scala/scala/meta/internal/mtags/CommonMtagsEnrichments.scala

```diff
@@ -42,21 +42,19 @@ trait CommonMtagsEnrichments {
       cls: java.lang.Class[T],
       gson: Option[Gson] = None
   ): Option[T] =
-    for {
-      data <- Option(obj)
-      value <-
-        try {
-          Option(
-            gson
-              .getOrElse(new Gson())
-              .fromJson[T](data.asInstanceOf[JsonElement], cls)
-          )
-        } catch {
-          case NonFatal(e) =>
-            logger.log(Level.SEVERE, s"decode error: $cls", e)
-            None
-        }
-    } yield value
+    Option(obj).flatMap { data =>
```

Comment:
Could you explain a bit more why this part was problematic? It seems the same code but without the for comp

Reply:
`import scala.language.unsafeNulls` has some limitations and this is one of it. I think the details can be found here. https://github.com/lampepfl/dotty/issues/18775#issuecomment-1784319925

The issue with for comprehension is that after desugar it runs the yield function in another map so:
```scala
for {
  x <- Option(test.fromJson[T]())
} yield (x)
```

gets desugared into

```scala
Option.apply(test.fromJson[T]()).map(x => x)
```
Thus, according to rules and what was described in dotty issue, as last method is not a direct call to Java method but rather an anonymous function `x => x` which is basically new symbol, thus the type checking won't be run another time. My changes ensure that Java method is run at the end, thus works with the rules of unsafe nulls.

Reply:
Should we also add explicit nulls here so that we don't have a regression in the future?

Reply:
I was thinking on how to do this, but basically it is not so trivial especially as we cross compile it to both 2 and 3, and what is more we use version 2.13 as a "dependency from source" in dotty. Explicit nulls wasn't present in Scala 2 afaik

Reply:
We do compile it under Scala 3 though, wouldn't it be enough to add it when the version is Scala 3?

Reply:
Yeah, but then we have to add the `import scala.language.unsafeNull` to every file to compile under Scala 3. And this flag is not present in scala 2. There is also another option to add `.nn` to every place where explicit nulls is applied but it is also not present for Scala 2.